service/raft/raft_group0_client.cc

```diff
@@ -293,4 +294,36 @@ group0_command raft_group0_client::prepare_command(table_query query) {
     return group0_cmd;
 }
 
+raft_group0_client::query_result_guard::query_result_guard(utils::UUID query_id, raft_group0_client& client)
+    : _query_id{query_id}, _client{&client} {
+    _client->_results.emplace(_query_id, std::nullopt);
+}
+
+raft_group0_client::query_result_guard::query_result_guard(raft_group0_client::query_result_guard&& other)
+    : _query_id{other._query_id}, _client{other._client} {
+    other._client = nullptr;
+}
+
+raft_group0_client::query_result_guard::~query_result_guard() {
+    if (_client != nullptr) {
+        _client->_results.erase(_query_id);
+    }
+}
+
+raft::broadcast_tables::query_result raft_group0_client::query_result_guard::get() const {
+    assert(_client->_results.contains(_query_id) && _client->_results[_query_id].has_value());
+
+    return std::move(*_client->_results[_query_id]);
```

Comment:
3 map searches! (two in assert, then the last here).
Use `unordered_map::find()` to reduce it to a single search.

Reply:
Also don't use `assert`, use `if`+`on_internal_error`.